2025-04-14-January model to send for review.nlogo

```diff
@@ -0,0 +1,1081 @@
+end
+
+to create-wheel-network
+  ;
+  let central-turtle number-turtles - 1
+
+  ; Create the ring among non-central scientists
+  ask scientists with [who != central-turtle] [
+    let target1 (who - 1) mod (count turtles - 1)
+    let target2 (who + 1) mod (count turtles)
+    if target1 = 0 [set target1 (count turtles - 1)]
+    if target2 = 0 [set target2 1]
+    create-Sci-Sci-with scientist (ifelse-value (target1 = 0) [(count turtles - 1)] [target1])
+    create-Sci-Sci-with scientist (ifelse-value (target2 > (count turtles - 1)) [1] [target2])
+  ]
+
+  ; Connect the central scientist to all other scientists
+  ask turtle central-turtle [
+    ask other scientists [create-link-with turtle central-turtle]
+  ]
+end
+
+to setup-complete
+  clear-turtles
+  reset-ticks
+  update-real-probabilities
+  ask patches [ set pcolor blue ]
+  create-scientists number-turtles
+  layout-circle sort scientists 8
+   ; Shift the circle to the right
+  let shift-x max-pxcor / 2 - 14 ; Adjust this to move the circle rightward, closer to the edge
+  ask scientists [
+    setxy xcor + shift-x ycor
+  ]
+  create-complete-network
+  ask scientists [
+    set shape "circle"
+    set color green
+    initialize-beliefs
+  ]
+end
+
+to create-complete-network
+  ; For each turtle, create a link to every other turtle.
+  ask scientists [
+    let current-turtle self ; Store a reference to the current turtle
+    ask scientists with [self != current-turtle] [
+      create-Sci-Sci-with current-turtle [
+      set color red ; Set the color of Sci-Sci links to blue
+    ]
+  ]
+    ]
+
+end
+
+; POLICYMAKER SETUP
+
+to setup-policymakers
+  ; Determine starting positions
+  let base-xcor max-pxcor - 5 ; Adjust to move policymakers further to the right if needed
+  let start-ycor 0 ; Adjust starting ycor to align with the middle of the world
+  let spacing 4 ; Adjust the spacing between policymakers as needed
+
+  ; Create policymakers in a vertical line
+  create-policymakers 5 [
+    set shape "square"
+    set color green
+    ; This will position them in a vertical line to the right
+    let my-index who - min [who] of policymakers ; Calculate relative index for vertical positioning
+    set xcor base-xcor
+    set ycor start-ycor + (my-index * spacing) - ((count policymakers - 1) / 2.0 * spacing)
+    set credence random-float .5
+    set label precision credence 3
+  ]
+  link-pol-to-sci
+end
+
+to setup-propagandists
+  create-propagandists 1 [
+    set shape "face neutral"
+    set color white
+    set size 4
+    ;; position the propagandist in the middle at the top
+    setxy 0 (max-pycor - 6)
+  ]
+  link-prop-to-sci
+  link-prop-to-pol
+end
+
+
+to link-pol-to-sci
+  ; For each policymaker, select policymaker-connectedness scientists and create links
+  ask policymakers [
+    let target-scientists n-of policymaker-connectedness scientists
+    print (word "Policymaker " who " is linking to " count target-scientists " scientists.")
```

Comment:
this print is relatively expensive, if performance is a consideration. try commenting out.